src/gui/QvisColorTableWindow.C

```diff
@@ -726,70 +906,133 @@ QvisColorTableWindow::UpdateEditor()
-                for(int i = 0; i < ctNames.size(); ++i)
+                int k = 0;
+                // go thru global tags
+                while (k < tagList.size())
                 {
-                    QTreeWidgetItem *treeItem = new QTreeWidgetItem();
-                    treeItem->setText(0, ctNames.at(i));
-                    treeGroup->addChild(treeItem);
+                    // if the global tag is active
+                    if (activeTags[k])
+                    {
+                        // and if the global tag is the same as our current local tag 
+                        if (tagList[k] == colorAtts->GetColorTables(i).GetTag(j))
+                        {
+                            // any tag was found
+                            anyTagFound = true;
+                            break;
+                        }
+                    }
+                    k ++;
                 }
-                nameListBox->addTopLevelItem(treeGroup);
-                ++iter;
+
+                // we only care if one tag was found
+                if (anyTagFound) break;
+                j ++;
             }
+
+            // if any tag was found, we add the color table to the list
+            if (anyTagFound)
+            {
+                QString item(colorAtts->GetNames()[i].c_str());
+                QTreeWidgetItem *treeItem = new QTreeWidgetItem(nameListBox);
+                treeItem->setText(0, item);
+                nameListBox->addTopLevelItem(treeItem);
+            }
+            colorAtts->SetActiveElement(i, anyTagFound);
         }
-        else
-        {
-            // revert to undecorated tree if there is only 1 category.
-            doSimpleTree = true;
-        }
-        mappedCT.clear();
     }
 
-    if(doSimpleTree)
+    // so tags must match all
+    if (doTags && !tagsMatchAny)
     {
         nameListBox->setRootIsDecorated(false);
-        for(int i = 0; i < colorAtts->GetNumColorTables(); ++i)
+        for (int i = 0; i < colorAtts->GetNumColorTables(); i ++)
         {
-            QString item(colorAtts->GetNames()[i].c_str());
-            QTreeWidgetItem *treeItem = new QTreeWidgetItem(nameListBox);
-            treeItem->setText(0, item);
-            nameListBox->addTopLevelItem(treeItem);
+            bool allTagsFound = true;
+            int j = 0;
+            // go thru global tags
+            while (j < tagList.size())
+            {
+                // if the global tag is active
+                if (activeTags[j])
+                {
+                    bool foundLocalTag = false;
+                    int k = 0;
+                    while (k < colorAtts->GetColorTables(i).GetNumTags())
+                    {
+                        if (tagList[j] == colorAtts->GetColorTables(i).GetTag(k))
+                        {
+                            foundLocalTag = true;
+                            break;
+                        }
+                        k ++;
```

Comment:
Ok, so seeing more instances of this now. It looks like this is a new sytlistic choice of yours and not a typo as I originally interpreted.

Honestly, its not my cup of tea. I never liked Brad's use of no space between `for` and `if` and the required paraentheses either. And, when I've cut-n-pasted his code, I've often changed it.

My personal practice is to follow *when in Rome do as the Romans* (meaning I try to stay consistent to what the rest of the code looks like even if it doesn't match my personal style) but I don't want to stall progress with stylistic debates either.

I thought we have a programmer's style guide but I can't seem to find it at the moment and even if I did, I don't think I'd find any verbiage there specifically speaking to this.

Reply:
Yes, that's typically how I've done things for the past few years at least. I always felt that it looked cleaner (much like how we both like the space between `for` and `if` and their parens) but I know not everyone is a fan. I will try, going forward, to match the style of the code I'm editing. It'll be a hard habit to break though. I think for this PR I'll leave these changes in since I do this _everywhere_ and time is of the essence.